gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/RepositoryFileCache.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions.intellij;
+
+import com.google.common.hash.Hashing;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RepositoryFileCache {
+    private static final Logger log = LoggerFactory.getLogger(RepositoryFileCache.class);
+
+    private static final String CACHE_PATH = System.getProperty("user.home") + "/.gcv-cache";
+
+    private static final Map<String, Set<String>> cache = new HashMap<>();
+
+    private final ExecutorService executorService = Executors.newFixedThreadPool(20);
+
+    public final void syncCache(String repoUrl, Set<String> packages) {
```

Comment:
It's also really tricky to do this properly in the presence of concurrent readers/writers.

Reply:
Interesting, I did have a look around and couldn't find any good packages todo this form me, in general do you think using a file cache rather than / as well as the gradle cache is worth it?

Reply:
Honestly, in this situation I'd consider using something like sqlite to make the cache as that will handle all the hard bits for you. But I would also question what this kind of caching will bring:
1. We'll still need to check every so often for new packages in case something has been newly added.
2. It's only going to cover all the cases that have been individually searched via the idea plugin rather than the Gradle cache just having everything ever used in Gradle builds.

Reply:
good point, I was trying to avoid the issue with gradle cache where we don't know the source of the package but I'm not sure how much of a real problem that is - given we mark the suggestions from the cache clearly